dashboard/test/controllers/plc/course_units_controller_test.rb

```diff
@@ -0,0 +1,32 @@
+require 'test_helper'
+
+class Plc::CourseUnitsControllerTest < ActionController::TestCase
+  setup do
+    @admin = create(:admin)
+
+    @plc_course = create(:plc_course)
+    @course_unit = create(:plc_course_unit, plc_course: @plc_course, unit_name: 'PLC Course')
+  end
+
+  # only admins can view the plc course launch page
+  test_user_gets_response_for :launch, params: -> {}, user: nil, response: :redirect, redirected_to: '/users/sign_in'
+  test_user_gets_response_for :launch, params: -> {}, user: :student, response: :forbidden
+  test_user_gets_response_for :launch, params: -> {}, user: :teacher, response: :forbidden
+  test_user_gets_response_for :launch, params: -> {}, user: :levelbuilder, response: :forbidden
+  test_user_gets_response_for :launch, params: -> {}, user: :admin, response: :success
+
+  # only admins can call launch plc course
+  test_user_gets_response_for :launch_plc_course, params: -> {{plc_course_unit_id: @course_unit.id}}, user: nil, response: :redirect, redirected_to: '/users/sign_in'
```

Comment:
Pretty impressed with the test coverage generally here, thank you! One question though -- shouldn't these be have a parameter `method: :post`?

Reply:
Hmm weird. It does seem like they worked but looking at the method more closely does seem like they should include the method. I will update